include/CppUTest/TestMemoryAllocator.h

```diff
@@ -96,5 +96,37 @@ class NullUnknownAllocator: public TestMemoryAllocator
     static TestMemoryAllocator* defaultAllocator();
 };
 
+
+struct LocationToFailAlloc
+{
+    int allocNumberToFail;
+    int actualAllocNumber;
+    const char* file;
+    int line;
+};
+
+class FailableMemoryAllocator: public TestMemoryAllocator
+{
+public:
+    enum {MAX_NUMBER_OF_FAILED_ALLOCS = 10};
+    FailableMemoryAllocator(const char* name_str = "failable alloc", const char* alloc_name_str = "alloc", const char* free_name_str = "free");
+    virtual ~FailableMemoryAllocator() {}
+    virtual void failAllocNumber(int number);
+    virtual void failNthAllocAt(int allocationNumber, const char* file, int line);
```

Comment:
I'd probably do file, line and make the allocationNumber to be a default parameter which defaults to 1